.github/deploy/steps/85-Create-And-Connect-Metastore.ps1

```diff
@@ -0,0 +1,132 @@
+
+Write-Host "    Get access token for Az Global Admin SPN to create Metastore"
+. $PSScriptRoot/15-ensure-keyvault-access.ps1
+$cicdSpn = Get-SpnWithSecret -spnName $cicdSpnName -keyVaultName $keyVaultName
+Throw-WhenError -output $cicdSpn
+
+Write-Host "    Add the SPN to the Databricks Workspace as an admin user"
+$metastoreBearerToken = Set-DatabricksSpnAdminUser `
+  -tenantId $tenantId `
+  -clientId $cicdSpn.clientId `
+  -clientSecret $cicdSpn.secretText `
+  -workspaceUrl $metastoreDatabricks.workspaceUrl `
+  -resourceId $metastoreDatabricks.id
+Throw-WhenError -output $metastoreBearerToken
+
+Write-Host "    Generate SPN personal access token"
+$metastoreDatabricksToken = ConvertTo-DatabricksPersonalAccessToken `
+  -workspaceUrl $metastoreDatabricks.workspaceUrl `
+  -bearerToken $metastoreBearerToken
+Throw-WhenError -output $metastoreDatabricksToken
+
+[Environment]::SetEnvironmentVariable('DATABRICKS_AAD_TOKEN', $metastoreDatabricksToken)
+
+Write-Host "    Connect"
+$metastoreDatabricksUrlHttps = "https://" + $metastoreDatabricks.workspaceUrl
+$output = databricks configure --host $metastoreDatabricksUrlHttps --aad-token
+Throw-WhenError -output $output
+
+Write-Host "  Create Metastore" -ForegroundColor DarkYellow
+
+$metastoreStorageAccountEndpoint = $metastoreStorageAccount.primaryEndpoints.dfs.TrimStart("https://")
+$metastoreContainerEndpoint = "abfss://${metastoreContainerName}@${metastoreStorageAccountEndpoint}meta"
+
+Write-Host "    Check if exists"
+$metastores = databricks unity-catalog metastores list | ConvertFrom-Json
+Throw-WhenError -output $metastores
+
+if ($metastores.metastores.name -notcontains $metastoreName) {
+  Write-Host "      Metastore does NOT exist. Creating..." -ForegroundColor DarkGray
+  $metastore = databricks unity-catalog metastores create `
+    --name $metastoreName `
+    --region $location `
+    --storage-root $metastoreContainerEndpoint `
+    | ConvertFrom-Json
+  Throw-WhenError -output $metastore
+  $metastoreId = $metastore.metastore_id
+} else {
+  Write-Host "      Metastore ${metastoreName} exists. Getting id..." -ForegroundColor DarkGray
+  $metastoreId = ($metastores.metastores | Where-Object { $_.name -eq $metastoreName }).metastore_id
+}
+
+Write-Host "  Assign workspace to metastore" -ForegroundColor DarkYellow
+Write-Host "    Check if the workspace is assigned to the metastore"
+$assignment = databricks unity-catalog metastores get-assignment
+
+if ($assignment.Contains("METASTORE_DOES_NOT_EXIST")) {
+  Write-Host "      Workspace is NOT assigned correctly. Assigning..." -ForegroundColor DarkGray
+  $output = databricks unity-catalog metastores assign `
+    --workspace-id $metastoreDatabricks.workspaceId `
+    --metastore-id $metastoreId
+  Throw-WhenError -output $output
+} else {
+  Write-Host "      Workspace is assigned correctly" -ForegroundColor DarkGray
+}
+
+Write-Host "  Add storage credential for access connector for workspace" -ForegroundColor DarkYellow
```

Comment:
could you provide a little comment here.
This part is a bit confusing if you are that into this.

// In order to connect the ....
// therefore it is neccesary to....

Reply:
Done